arch/arc/core/cache.c

```diff
@@ -110,36 +95,37 @@ static void dcache_flush_mlines(uint32_t start_addr, uint32_t size)
 				break;
 			}
 		} while (1);
-		start_addr += DCACHE_LINE_SIZE;
-	} while (start_addr <= end_addr);
+		start_addr_temp += DCACHE_LINE_SIZE;
```

Comment:
start_addr_temp is increased by d-cache line size.  Does line 87 needs to be changed accordingly? 		`z_arc_v2_aux_reg_write(_ARC_V2_DC_FLDL, start_addr);` --> `z_arc_v2_aux_reg_write(_ARC_V2_DC_FLDL, start_addr_temp); `with current logic? [please follow @stephanosio's suggestion above, then this change shouldn't be required]

Reply:
> Would like to move forward with this PR. Improvements can be taken up later.

Have a comment to be addressed first..